olclient/entity_helpers/work.py

```diff
@@ -138,6 +139,17 @@ def rm_subjects(self, subjects, comment=''):
             data['subjects'] = list(set(data['subjects']) - set(subjects))
             return self.OL.session.put(url, json.dumps(data))
 
+        def delete(self, comment: str, confirm: bool = True):
+            edition_olids: List[str] = [edition_data.olid for edition_data in self.editions]
+            edition_count = len(edition_olids)
+            should_delete = confirm is False or get_approval_from_cli(
+                f'Delete https://openlibrary.org/works/{self.olid} and its {edition_count} editions?'
```

Comment:
Oh, and should probably specify what the user needs to type :P
```suggestion
                f'Delete https://openlibrary.org/works/{self.olid} and its {edition_count} editions? (y/n)'
```